src/main/java/org/testng/internal/Parameters.java

```diff
@@ -43,6 +38,42 @@
  */
 public class Parameters {
   public static final String NULL_VALUE= "null";
+  private static Class<?>[] annotationList = new Class<?>[] {
+      BeforeSuite.class,
+      AfterSuite.class,
+      BeforeTest.class,
+      AfterTest.class,
+      BeforeClass.class,
+      AfterClass.class,
+      BeforeGroups.class,
+      AfterGroups.class,
+      BeforeMethod.class,
+      AfterMethod.class
+  };
+
+  private static Map<String, List<Class>> mapping = new HashMap<>();
+
+  static {
+    List<Class> ctxTest = Arrays.asList(new Class[] {ITestContext.class, XmlTest.class});
```

Comment:
This cannot be done, because I am being thrown errors by IntelliJ wherein for the first statement 
`List<Class<?>> ctxTest = Arrays.asList(ITestContext.class, XmlTest.class);` it says that expected was `List<?>` but actual values were `List<? extends Serializable>`
and for the second one, there's a Object[] array. Its difficult to maintain parity across both.

So I chose to leave this as is and Not address this comment.